src/commands/files.ts

```diff
@@ -1,13 +1,45 @@
 import { promises } from 'fs'
 
+// the maximum number of attempts at finding a unique filename (by appending a suffix).
+const MAX_ATTEMPTS = 100
+
+/**
+ * Writes content to file but does not overwrite any existing file with the same name.
+ * If a file with the given name already exists, attempts to write to the file by
+ * appending a suffix. For example, if filename "foo.txt" already exists, it will attempt to
+ * write the file as "foo.txt.1" then "foo.txt.2" until it finds an unused filename.
```

Comment:
Why not `foo1.txt` instead? Does it matter that the filename extension is no longer at the end?

Reply:
Because I'm lazy and didn't want to mess with file name parsing and extensions.